src/array/ops/image.rs

```diff
@@ -371,6 +398,25 @@ impl ImageArray {
         Some(result)
     }
 
+    pub fn encode(&self, image_format: ImageFormat) -> DaftResult<BinaryArray> {
+        let result = (0..self.len())
+            .map(|i| self.as_image_obj(i))
+            .map(|img| {
+                img.map(|img| {
+                    let mut buf = Vec::new();
+                    img.encode(image_format, &mut buf)?;
```

Comment:
I should have clarified better but I was thinking we have the `img.encode` extend single buffer that is passed in rather than allocate a new buffer for each image and then concat it after